duckduckgo/Android
# Feature/custom url schemes

Task/Issue URL: https://app.asana.com/0/488551667048375/656818431539163
GitHub Issue: https://github.com/duckduckgo/Android/issues/288

Tech Design URL: 
CC: 

**Description**:
Detect custom schemes and determine if an external app can open it or not. If an app is installed which can handle it, we'll prompt the user to ensure they are happy with jumping outside of our app before launching an external app.

**Steps to test this PR**:
1. Search for "uber google play" 
1. Visit the search result for the Google Play Store
1. Click the floating button for "OPEN IN PLAY STORE APP"
1. Test the positive and negative buttons for the dialog which is shown

![device-2018-08-07-144630](https://user-images.githubusercontent.com/1336281/43779627-b871d6e8-9a50-11e8-8f0b-e24fc7b770c6.png)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
